uninstall.sh

```diff
@@ -0,0 +1,388 @@
+else
+  tty_escape() { :; }
+fi
+tty_mkbold() { tty_escape "1;${1:-39}"; }
+tty_blue=$(tty_mkbold 34)
+tty_red=$(tty_mkbold 31)
+tty_bold=$(tty_mkbold 39)
+tty_reset=$(tty_escape 0)
+
+have_sudo_access() {
+  if [[ -z "${HAVE_SUDO_ACCESS-}" ]]; then
+    /usr/bin/sudo -l mkdir &>/dev/null
+    HAVE_SUDO_ACCESS="$?"
+  fi
+
+  if [[ -z "${HOMEBREW_ON_LINUX-}" ]] && [[ "$HAVE_SUDO_ACCESS" -ne 0 ]]; then
+    abort "Need sudo access on macOS!"
+  fi
+
+  return "$HAVE_SUDO_ACCESS"
+}
+
+shell_join() {
+  local arg
+  printf "%s" "$1"
+  shift
+  for arg in "$@"; do
+    printf " "
+    printf "%s" "${arg// /\ }"
+  done
+}
+
+resolved_pathname() { realpath "$1"; }
+
+pretty_print_pathnames() {
+  local p
+  for p in "$@"; do
+    if [[ -h $p ]]; then
+      printf '%s -> %s\n' "$p" "$(resolved_pathname "$p")"
+    elif [[ -d $p ]]; then
+      echo "$p/"
+    else
+      echo "$p"
+    fi
+  done
+}
+
+chomp() {
+  printf "%s" "${1/"$'\n'"/}"
+}
+
+ohai() {
+  printf "${tty_blue}==>${tty_bold} %s${tty_reset}\n" "$(shell_join "$@")"
+}
+
+warn() {
+  printf "${tty_red}Warning${tty_reset}: %s\n" "$(chomp "$1")"
+}
+
+execute() {
+  if ! "$@"; then
+    abort "$(printf "Failed during: %s" "$(shell_join "$@")")"
+  fi
+}
+
+execute_sudo() {
+  local -a args=("$@")
+  if [[ -n "${SUDO_ASKPASS-}" ]]; then
+    args=("-A" "${args[@]}")
+  fi
+  if have_sudo_access; then
+    ohai "/usr/bin/sudo" "${args[@]}"
+    system "/usr/bin/sudo" "${args[@]}"
+  else
+    ohai "${args[@]}"
+    system "${args[@]}"
+  fi
+}
+
+system() {
+  if ! "$@"; then
+    warn "Failed during: $(shell_join "$@")"
+    failed=true
+  fi
+}
+
+####################################################################### script
+
+homebrew_prefix_candidates=()
+
+usage() {
+  cat <<EOS
+Homebrew Uninstaller
+Usage: $0 [options]
+    -p, --path=PATH  Sets Homebrew prefix. Defaults to /usr/local.
+        --skip-cache-and-logs
+                     Skips removal of HOMEBREW_CACHE and HOMEBREW_LOGS.
+    -f, --force      Uninstall without prompting.
+    -q, --quiet      Suppress all output.
+    -d, --dry-run    Simulate uninstall but don't remove anything.
+    -h, --help       Display this message.
+EOS
+  exit "${1:-0}"
+}
+
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -p*) homebrew_prefix_candidates+=("${1#-p}");;
+    --path=*) homebrew_prefix_candidates+=("${1#--path=}");;
+    --skip-cache-and-logs) opt_skip_cache_and_logs=1;;
+    -f|--force) opt_force=1;;
+    -q|--quiet) opt_quiet=1;;
+    -d|--dry-run) opt_dry_run=1;;
+    -h|--help) usage;;
+    *) warn "Unrecognized option: '$1'"; usage 1;;
+  esac
+  shift
+done
+
+if [[ ${#homebrew_prefix_candidates[@]} -eq 0 ]]; then # Attempt to locate Homebrew unless `--path` is passed
+  prefix=$(brew --prefix)
+  [[ -n $prefix ]] && homebrew_prefix_candidates+=("$prefix")
+  # shellcheck disable=SC2230  # "which" is a legitimate fallback
+  prefix=$(command -v brew) || prefix=$(which brew) || prefix=""
```

Comment:
I see this is a direct translation from the original code, but I'm not sure we still need this fallback code (`command -v brew` fails but `which brew` succeed?). In ruby it invokes the executable (maybe `/usr/bin/command`), but in bash it's a builtin function.

Reply:
Good point, will fix.